File: sample/src/main/java/com/thoughtbot/expandablerecyclerview/sample/singlecheck/SingleCheckGenre.java
package com.thoughtbot.expandablerecyclerview.sample.singlecheck;

import android.os.Parcel;
import com.thoughtbot.expandablecheckrecyclerview.models.SingleCheckExpandableGroup;
import java.util.List;

public class SingleCheckGenre extends SingleCheckExpandableGroup {

  private int iconResId;

  public SingleCheckGenre(String title, List items, int iconResId) {
    super(title, items);
    this.iconResId = iconResId;
  }

  protected SingleCheckGenre(Parcel in) {
    super(in);
    iconResId = in.readInt();
  }

  public int getIconResId() {
    return iconResId;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    super.writeToParcel(dest, flags);
    dest.writeInt(iconResId);
  }

  @Override
  public int describeContents() {
    return 0;
  }

  public static final Creator<SingleCheckGenre> CREATOR = new Creator<SingleCheckGenre>() {
    @Override
    public SingleCheckGenre createFromParcel(Parcel in) {
      return new SingleCheckGenre(in);
    }

    @Override
    public SingleCheckGenre[] newArray(int size) {
      return new SingleCheckGenre[size];
    }
  };
}

File: sample/src/main/java/com/thoughtbot/expandablerecyclerview/sample/singlecheck/SingleCheckGenre.kt
package com.thoughtbot.expandablerecyclerview.sample.singlecheck

import android.os.Parcel
import android.os.Parcelable
import com.thoughtbot.expandablecheckrecyclerview.models.SingleCheckExpandableGroup

class SingleCheckGenre : SingleCheckExpandableGroup {
    var iconResId: Int
        private set

    constructor(title: String?, items: List<*>?, iconResId: Int) : super(title, items) {
        this.iconResId = iconResId
    }

    protected constructor(`in`: Parcel) : super(`in`) {
        iconResId = `in`.readInt()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
        dest.writeInt(iconResId)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<SingleCheckGenre?> = object : Parcelable.Creator<SingleCheckGenre?> {
            override fun createFromParcel(`in`: Parcel): SingleCheckGenre? {
                return SingleCheckGenre(`in`)
            }

            override fun newArray(size: Int): Array<SingleCheckGenre?> {
                return arrayOfNulls(size)
            }
        }
    }
}